### PR TITLE
Add npm package for function-runner

### PR DESCRIPTION
**Pull request by surma**
This is the exact same JS code as we use for Javy, just with some naming adjustments.

I'd probably want to refactor this “grab a binary from github” code into a library, but I wanted to get this into place first.

### CODE REVIEW

File: npm/function-runner/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+```
+# Install function-runner globally
+$ npm install -g function-runner
+
+# Directly invoke it via npm
+$ npx function-runner
+```
+
+## Updating function-=runner
```
Comment (saulecabrera) [inline npm/function-runner/README.md:17]:
```suggestion
## Updating function-runner
```

File: npm/function-runner/index.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+			`git clone https://github.com/shopify/function-runner ${repoDir}`
+		);
+		console.log("Building function-runner...");
+		childProcess.execSync("cargo build --release", { cwd: repoDir });
+	} catch (e) {
+		console.error(e);
+		console.error("");
+		console.error("BUILDING FUNCTION-RUNNER FAILED");
+		console.error("Please make sure you have Rust installed");
+		console.error("See the javy README for more details.");
```
Comment (saulecabrera) [inline npm/function-runner/index.js:139]:
```suggestion
                console.error("See the function-runner README for more details.");
```

File: npm/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+	"useTabs": true
```
Comment (saulecabrera) [inline npm/.prettierrc:2]: Why is this preferred over 2 spaces?

**Review by saulecabrera — APPROVED** [review]
Some minor changes needed, else good to merge.

File: npm/.prettierrc (hunk 3 above)
Comment (surma) [reply npm/.prettierrc:2]: [Tabs are significantly better than spaces wrt a11y](https://twitter.com/DasSurma/status/1553634641491509248) :)

**Pull request merged**